kalamine/template.py

```diff
@@ -387,18 +395,206 @@ def klc_deadkeys(layout: "KeyboardLayout") -> List[str]:
+                dead_symbols.append(dead)
+            else:
+                desc = " "
+                symbols.append("WCH_NONE")
+                dead_symbols.append("WCH_NONE")
+
+        virtual_key = klc_virtual_key(layout, symbols, "")
+        if len(virtual_key) == 1:
+            virtual_key = "'" + virtual_key + "'"
+        else:
+            virtual_key = "VK_" + virtual_key
+
+        def process_symbol(symbol: str) -> str:
+            if len(symbol) == 4:
+                return "0x" + symbol
+            elif len(symbol) == 1:
+                return "'" + symbol + "'"
+            else:
+                return symbol
+
+        symbols[:] = [process_symbol(symbol) for symbol in symbols]
+        dead_symbols[:] = [process_symbol(symbol) for symbol in dead_symbols]
+
+        if layout.has_altgr:
+            output.append(
+                "\t{"
+                + "\t,".join(
+                    [
+                        virtual_key,
+                        "CAPLOK" if alpha else "0",  # affected by CapsLock?
+                        symbols[0],
+                        symbols[1],  # base layer
+                        "WCH_NONE",
+                        "WCH_NONE",  # ctrl layer
+                        symbols[2],
+                        symbols[3],  # altgr layer
+                    ]
+                )
+                + "},"
+            )
+            if has_dead:
+                output.append(
+                    "\t{"
+                    + "\t,".join(
+                        [
+                            "0xff",
+                            "0",
+                            dead_symbols[0],
+                            dead_symbols[1],
+                            "WCH_NONE",
+                            "WCH_NONE",  # ctrl layer
+                            dead_symbols[2],
+                            dead_symbols[3],
+                        ]
+                    )
+                    + "},"
+                )
+        else:
+            output.append(
+                "\t{"
+                + "\t".join(
+                    [
+                        virtual_key,
+                        "CAPLOK" if alpha else "0",  # affected by CapsLock?
+                        symbols[0],
+                        symbols[1],  # base layer
+                        "WCH_NONE",
+                        "WCH_NONE",  # ctrl layer
+                    ]
+                )
+                + "},"
+            )
+            if has_dead:
+                output.append(
+                    "\t{"
+                    + "\t,".join(
+                        [
+                            "0xff",
+                            "0",
+                            dead_symbols[0],
+                            dead_symbols[1],
+                            "WCH_NONE",
+                            "WCH_NONE",  # ctrl layer
+                        ]
+                    )
+                    + "},"
+                )
```

Comment:
This could probably be factorized by using a function like :

```python
               def key_list(key_syms, virtual_key="0xff", has_altgr=False, alpha=False) -> str:
                    cols = [
                        virtual_key,
                        "CAPLOK" if alpha else "0",  # affected by CapsLock?
                        key_syms[0],
                        key_syms[1],  # base layer
                        "WCH_NONE",
                        "WCH_NONE",  # ctrl layer
                    ]
                    if has_altgr:
                        cols += [
                            symbols[2],
                            symbols[3],  # altgr layer
                        ]
                    return "\t,".join(cols)